test/behavior/steps/login.py

```diff
@@ -0,0 +1,81 @@
+from behave import given, when, then
+
+
+keyword_dict = {inconsequential_input: 'something',
+				incorrect_input: 'incorrect'}
+
+
+@given('User is at the Tendrl login page')
+def step_impl(context):
+	context.browser.get(context.data['common']['tendrl']['ip'])
+
+
+@given('Tendrl API is not reachable')
+def step_impl(context):
+	pass
+
+
+@given('User is at the "cluster" list page')
+def step_impl(context):
+	context.browser.get(context.data['common']['tendrl']['ip'])
+	assert "cluster" in context.browser.current_url
+
+
+@when('User types in a correct "{field}"')
+def step_impl(context, field):
+	context.browser.find_element_by_id(field).send_keys(context.data['common']['tendrl'][field])
+
+
+@when('User types in an incorrect "{field}"')
+def step_impl(context, field):
+	context.browser.find_element_by_id(field).send_keys(keyword_dict[incorrect_input])
+
+
+@when('User types something in the "{field}" field')
+def step_impl(context, field):
+	context.browser.implicitly_wait(3)
+	context.browser.find_element_by_id(field).send_keys(keyword_dict[inconsequential_input])
+
+
+@when('User deletes what is in the "{field}" field')
+def step_impl(context, field):
+	context.browser.find_element_by_id(field).clear()
+
+
+@when('User clicks the Login button')
+def step_impl(context):
+	context.browser.find_element_by_class_name("login-btn").click()
+
+
+@when('User clicks on the Admin dropdown')
+def step_impl(context):
+	context.browser.find_element_by_id('usermenu').click()
+
+
+@when('User clicks on the Logout button')
```

Comment:
make it generic

Reply:
The generic function uses class to find elements but this won't work for the Logout button as it is part of a class and instead uses a unique ID, so I think it may be best to leave it nongeneric for now

Reply:
I am in favor of making all the buttons in tendrl ui to be generic so we don't need to write these tests for every button on UI. We can discuss this on friday.

Reply:
Most elements can be done with a generic function because our generic function uses class to find elements, but the HTML will not allow this for all buttons currently in tendrl/ui as some cannot be identified by class. We can go over the HTML together at some point to see if we can change this, but the effort to make all those changes would I believe be best suited as a separate PR, by which point we could retroactively change some of these functions.